### PR TITLE
Remove unnecessary conditional defines from mapkit.cs

### PR DESCRIPTION
**Pull request by chamons**
- Unlike the rest of my recent work, this is all done by hand, so please review closely.
- Part of https://github.com/xamarin/xamarin-macios/issues/14444

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

File: src/mapkit.cs [hunk 1]
```diff
@@ -1050,9 +1061,9 @@ interface MKPointAnnotation : MKGeoJsonObject {
 
 		[Export ("coordinate")]
 		CLLocationCoordinate2D Coordinate { get; set; }
-}
+	}
```
Comment (tj-devel709) [inline src/mapkit.cs:1064]: nit: spacing diff

File: src/mapkit.cs [hunk 2]
```diff
@@ -1521,18 +1544,17 @@ partial interface MKMapSnapshot {
 		[Export ("pointForCoordinate:")]
 		CGPoint PointForCoordinate (CLLocationCoordinate2D coordinate);
 
-#if MONOMAC
 		[NoWatch][NoTV][NoiOS]
```
Comment (tj-devel709) [inline src/mapkit.cs:1547]: Maybe a [NoMacCatalyst] should go here

Comment (chamons) [reply src/mapkit.cs:1547]: Nope, and xtro locally agrees. I had to remove some as they were used as a base class of something that is on catalyst.

**Review by tj-devel709 — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Comment by chamons** [comment]
Test failures unrelated https://github.com/xamarin/xamarin-macios/pull/14587

**Pull request merged**